app/jobs/transfer_basespace_files.rb

```diff
@@ -18,15 +18,19 @@ def self.perform(sample_id, basespace_dataset_id, basespace_access_token)
     if samples_remaining.empty?
       begin
         BasespaceHelper.revoke_access_token(basespace_access_token)
-      rescue
+      rescue HttpHelper::HttpError => e
         # If revoke_access_token is called multiple times, it will fail on the subsequent calls. This is okay.
-        nil
+        if e.status_code == 401
+          Rails.logger.warn("Revoke access token failed for sample #{sample_id}. Likely called multiple times by samples sharing the same token.")
+        else
+          raise e
+        end
       end
 
       # This verification call will work even if called multiple times.
       BasespaceHelper.verify_access_token_revoked(basespace_access_token, sample_id)
     end
   rescue => e
-    Rails.logger.error(e)
+    LogUtil.log_err_and_airbrake("Error transferring basespace files for sample #{sample_id}. Reason: #{e}")
```

Comment:
I would re-raise the error after logging it to airbrake, so we can eventually restart the failing task using resque (assuming this is being executed by a resque worker)